src/epoll.rs

```diff
@@ -151,41 +150,44 @@ impl Poller {
     pub fn wait(&self, events: &mut Events, timeout: Option<Duration>) -> io::Result<()> {
         log::trace!("wait: epoll_fd={}, timeout={:?}", self.epoll_fd, timeout);
 
-        // Configure the timeout using timerfd.
-        let new_val = libc::itimerspec {
-            it_interval: TS_ZERO,
-            it_value: match timeout {
-                None => TS_ZERO,
-                Some(t) => libc::timespec {
-                    tv_sec: t.as_secs() as libc::time_t,
-                    tv_nsec: (t.subsec_nanos() as libc::c_long).into(),
+        if let Some(timer_fd) = self.timer_fd {
+            // Configure the timeout using timerfd.
+            let new_val = libc::itimerspec {
+                it_interval: TS_ZERO,
+                it_value: match timeout {
+                    None => TS_ZERO,
+                    Some(t) => libc::timespec {
+                        tv_sec: t.as_secs() as libc::time_t,
+                        tv_nsec: (t.subsec_nanos() as libc::c_long).into(),
+                    },
                 },
-            },
-        };
-        syscall!(timerfd_settime(self.timer_fd, 0, &new_val, ptr::null_mut()))?;
+            };
 
-        // Set interest in timerfd.
-        self.interest(
-            self.timer_fd,
-            Event {
-                key: crate::NOTIFY_KEY,
-                readable: true,
-                writable: false,
-            },
-        )?;
+            syscall!(syscall(libc::SYS_timerfd_settime, timer_fd, 0, &new_val, 0))?;
```

Comment:
Let's be explicit with types to make sure we don't call the syscall with arguments of incorrect types:
```suggestion
            syscall!(syscall(libc::SYS_timerfd_settime, timer_fd, 0 as libc::c_int, &new_val as *const libc::itimerspec, 0 as *mut libc::itimerspec))?;
```